FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/CustomFacetGatewayTest.php

```diff
@@ -0,0 +1,416 @@
+            ->willReturn(
+                $responseCode !== null ? new \Zend_Http_Response(404, [], $xmlResponse->asXML()) : null
+            );
+
+        $mockHydrator = $this->getMockBuilder(
+            '\FinSearchUnified\Bundle\StoreFrontBundle\Gateway\Findologic\Hydrator\CustomListingHydrator'
+        )
+            ->setMethods(['hydrateFacet'])
+            ->getMock();
+        $mockHydrator->expects($this->never())
+            ->method('hydrateFacet');
+        $facetGateway = new CustomFacetGateway(
+            $mockOriginalService,
+            $mockHydrator,
+            $mockUrlBuilder
+        );
+
+        $facetGateway->getList([3], $context);
+    }
+
+    /**
+     * @return array
+     */
+    public function getListProviderForFindologicSearch()
+    {
+        return [
+            'FINDOLOGIC search is triggered and response is OK' => [
+                [
+                    ['name' => 'price', 'display' => 'Preis', 'select' => 'single', 'type' => 'range-slider']
+                ],
+                [
+                    [
+                        'name' => 'price',
+                        'uniqueKey' => 'price',
+                        'attribute_label' => 'Preis',
+                        'attribute_name' => 'product_attribute_price',
+                        'attribute_formfield_name' => 'price',
+                        'attribute_mode' => ProductAttributeFacet::MODE_RANGE_RESULT,
+                    ]
+                ]
+            ],
+            'No facets are returned' => [
+                [],
+                []
+            ],
+            'Single CustomFacet with Price' => [
+                [
+                    ['name' => 'price', 'display' => 'Preis', 'select' => 'single', 'type' => 'range-slider']
+                ],
+                [
+                    [
+                        'name' => 'price',
+                        'uniqueKey' => 'price',
+                        'attribute_label' => 'Preis',
+                        'attribute_name' => 'product_attribute_price',
+                        'attribute_formfield_name' => 'price',
+                        'attribute_mode' => ProductAttributeFacet::MODE_RANGE_RESULT,
+                    ]
+                ]
+            ],
+            'Two CustomFacet with Price and Color' => [
+                [
+                    ['name' => 'price', 'display' => 'Preis', 'select' => 'single', 'type' => 'range-slider'],
+                    ['name' => 'color', 'display' => 'Farbe', 'select' => 'multiple', 'type' => 'label'],
+                ],
+                [
+                    [
+                        'name' => 'price',
+                        'uniqueKey' => 'price',
+                        'attribute_label' => 'Preis',
+                        'attribute_name' => 'product_attribute_price',
+                        'attribute_formfield_name' => 'price',
+                        'attribute_mode' => ProductAttributeFacet::MODE_RANGE_RESULT,
```

Comment:
Since you already have all the data in `$filterData`, you could drop everything and only provide `attribute_mode` here. This way we would have less data to maintain.